src/__tests__/cat-card.test.tsx

```diff
@@ -0,0 +1,45 @@
+import { cleanup, render, screen } from '@testing-library/react';
+import { fetchCatImageMock } from './mocks/fetch-cat-image.mock';
+import { CatCard } from '../components/cat-card/cat-card';
+import { fakeCat } from './mocks/cats.mock';
+import { fakeCatImg } from './mocks/cat-image.mock';
+import type { Mock } from 'vitest';
+
+describe('Cat card', () => {
+  afterEach(() => {
+    cleanup();
+    vi.clearAllMocks();
+  });
+
+  it('should render a cat card with link', () => {
+    render(<CatCard cat={fakeCat} />);
+
+    const catLink = screen.getByRole('link', { name: fakeCat.name });
+
+    expect(catLink).toHaveAttribute('href', fakeCat.wikipedia_url);
+  });
```

Comment:
fixing react warning about state updates not being included in act(...):
```suggestion
  it('should render a cat card with link', () => {
    render(<CatCard cat={fakeCat} />);
    await waitFor(() => {
       const catLink = screen.getByRole('link', { name: fakeCat.name });
       expect(catLink).toHaveAttribute('href', fakeCat.wikipedia_url);
    });
  });
```